bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/HistoryTextWrapper.java

```diff
@@ -0,0 +1,207 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.ui.internal.findandreplace.overlay;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.DisposeEvent;
+import org.eclipse.swt.events.DisposeListener;
+import org.eclipse.swt.events.FocusListener;
+import org.eclipse.swt.events.KeyListener;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.widgets.TableItem;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.ToolItem;
+import org.eclipse.swt.widgets.Widget;
+
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+
+import org.eclipse.ui.internal.findandreplace.FindReplaceMessages;
+import org.eclipse.ui.internal.findandreplace.HistoryStore;
+
+/**
+ * Wrap a Text Bar and a ToolItem to add an input history. The text is only
+ * stored to history when requested by the Client code, the history is stored in
+ * a {@code HistoryStore} provided by the client. The history bar behaves like a
+ * normal {@code Text}.
+ */
+public class HistoryTextWrapper extends Composite {
+	private Text textBar;
+	private AccessibleToolBar tools;
+	private ToolItem dropDown;
+	private SearchHistoryMenu menu;
+	private HistoryStore history;
```

Comment:
nitpicky: all these fields (except for `menu`) can be `final`

Reply:
Yes, they can!
Fixed in https://github.com/eclipse-platform/eclipse.platform.ui/pull/1990/files#diff-5f8b29df540e278eb662337e8f6648429250ca274cd72b5e844ff86f13559f43R45